### ISSUE TITLE
Midpoint Review - Accessibility Feedback - Appointments Team, Appointments, Oracle Health - Direct Schedule and Request for an Appointment

### ISSUE THREAD

**Issue by briandeconinck** [department-of-veterans-affairs/va.gov-team#97728]
## Next Steps for the VFS team

- [ ] **Assign** this ticket to the team member(s) responsible for addressing feedback provided by Platform.
- [ ] **Comment** on this ticket:
  - If the Platform reviewer has any **Thoughts/Questions** that require responses.
  - When **Must** feedback has been incorporated. As appropriate, link to any other GitHub issues or PRs related to this feedback.
  - When **Should/Consider** feedback has been incorporated, or if any feedback will not be addressed. As appropriate, link to any other GitHub issues or PRs related to this feedback.
- [ ] **Questions?** For the most timely response, comment on Slack in your team channel tagging `@platform-governance-team-members`.
- [ ] **Close the ticket** when all feedback has been addressed.

## Thoughts/questions

- Sorry for the very late feedback! I was out sick most of last week and it's taken a while to dig out from that.
- Overall this looks good! No major concerns.

## Feedback
Practice areas will document their feedback on the VFS-provided artifacts following the [Must, Should, and Consider Framework](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/must-should-and-consider-framework-for-feedback). Platform Governance reviewers may also provide additional notes that don’t comment on the artifacts themselves but are important for implementation (eg. engineering/coding notes).

- [x] **Must:** Be careful about link text. When there are [multiple providers](https://www.figma.com/design/EHR9zBpGnAgc7bHqeIGEoT/OH-Direct-Schedule-MVP-%7C-Appointments-FE?node-id=5824-33201&t=tY8aNBgysgTqzq1k-4), each "Choose your preferred date and time" needs to have a unique accessible name. The easiest option here is for the visible link text to be unique, eg. "Choose your preferred date and time with [name]." If necessary, you also have the option of using an `aria-label` for screen reader users, eg. `<a href="..." aria-label="Choose your preferred date and time with [name]">Choose your preferred date and time</a>`.
- [x] **Must:** Be careful about heading hierarchy. From the visual presentation it looks like everything is nested appropriately, but it's worth outlining the heading levels across some of the more information-dense screens just to make sure everything is structured in a way that makes sense.

## Governance team actions
- [x] Format feedback as individual tasks (check boxes)
- [x] Assign this ticket to the VFS team member that opened the Slack request
- [x] Add the VFS team product label
- [x] Add the VFS team the feature label (if applicable)
- [x] Add the touchpoint labels
- [x] Add the practice area labels
- [x] Add the Collaboration Cycle initiative milestone

**Comment by melissa-lefevre** [comment]
> Must: Be careful about heading hierarchy. From the visual presentation it looks like everything is nested appropriately, but it's worth outlining the heading levels across some of the more information-dense screens just to make sure everything is structured in a way that makes sense.

Here are the semantic HTML headings
![Image](https://github.com/user-attachments/assets/4858d819-9d1c-4cd4-97b3-0ee17f96d75a)

**Comment by melissa-lefevre** [comment]
> Must: Be careful about link text. When there are [multiple providers](https://www.figma.com/design/EHR9zBpGnAgc7bHqeIGEoT/OH-Direct-Schedule-MVP-%7C-Appointments-FE?node-id=5824-33201&t=tY8aNBgysgTqzq1k-4), each "Choose your preferred date and time" needs to have a unique accessible name. The easiest option here is for the visible link text to be unique, eg. "Choose your preferred date and time with [name]." If necessary, you also have the option of using an aria-label for screen reader users, eg. <a href="..." aria-label="Choose your preferred date and time with [name]">Choose your preferred date and time</a>.

Insteadof using ARIA labes, we are using aria-describedby so that it's still usable by voice control users. Essentially an ID would be set up that pulls in the doctor's name from the header and adds it after the link phrase as supplemental context (ex. "Choose your preferred date and time with Julie Carson, RD").  This is per @rileyorr suggestion. 

![Image](https://github.com/user-attachments/assets/9cb7b19e-efad-4df8-aa22-52e0297e2bc1)

**Comment by rileyorr** [comment]
@melissa-lefevre thanks for looping me in! @briandeconinck I agree that the easiest solution would be to have unique links using the format "Choose your preferred date and time with [name]." and recommended that treatment as the most accessible option. However, I also mentioned that if the team believed that wasn't a feasible solution for this design, such as due to links becoming extremely long for example, then they should use aria-describedby. Bobby and I had noticed recently that Voice Control is unable to interact with duplicate links that are overwritten with aria-label since the exact aria-label needs to be spoken which is hidden unless the page is inspected. Happy to pivot my recommendation if aria-label should be what the team moves forward with!
